cuml/nearest_neighbors_mnmg_demo.ipynb

```diff
@@ -0,0 +1,376 @@
+    "\n",
+    "For additional information on cuML's MNMG nearest neighbors implementation: https://rapidsai.github.io/projects/cuml/en/latest/api.html#nearest-neighbors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "import pandas as pd\n",
+    "import cudf as gd\n",
+    "\n",
+    "from cuml.dask.common import to_dask_df\n",
+    "from cuml.dask.datasets import make_blobs\n",
+    "\n",
+    "from dask.distributed import Client, wait\n",
+    "from dask_cuda import LocalCUDACluster\n",
+    "\n",
+    "from sklearn.neighbors import NearestNeighbors as skNeighbors\n",
+    "from cuml.dask.neighbors import NearestNeighbors as cumlNeighbors"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Start Dask Cluster\n",
+    "\n",
+    "We can use the `LocalCUDACluster` to start a Dask cluster on a single machine with one worker mapped to each GPU. This is called one-process-per-GPU (OPG). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/share/conda/cuml_11/lib/python3.7/site-packages/distributed/dashboard/core.py:72: UserWarning: \n",
+      "Port 8787 is already in use. \n",
+      "Perhaps you already have a cluster running?\n",
+      "Hosting the diagnostics dashboard on a random port instead.\n",
+      "  warnings.warn(\"\\n\" + msg)\n",
+      "bokeh.server.util - WARNING - Host wildcard '*' will allow connections originating from multiple (or possibly all) hostnames or IPs. Use non-wildcard values to restrict access explicitly\n"
+     ]
+    }
+   ],
+   "source": [
+    "cluster = LocalCUDACluster(threads_per_worker=1)\n",
+    "client = Client(cluster)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define Parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_samples = 100000\n",
+    "n_features = 50\n",
+    "\n",
+    "n_total_partitions = len(list(client.has_what().keys()))\n",
+    "\n",
+    "n_neighbors = 5\n",
+    "\n",
+    "n_query = 5000"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Data\n",
+    "\n",
+    "### Device\n",
+    "\n",
+    "We can generate a dask_cudf.DataFrame of synthetic data for multiple clusters using `cuml.dask.datasets.make_blobs`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generating 50000 samples acgraross 2 partitions on 2 workers (total=100000 samples)\n"
+     ]
+    }
+   ],
+   "source": [
+    "X_cudf_train, _ = make_blobs(n_samples, \n",
```

Comment:
should probably fix the seed? do we want both sets of blobs to have the same centers (in which case, it would be nice to be explicit and pass in centers or seeds)